### DISCUSSION TITLE
easyhid couldnt load hidapi with ctypes.util.find_library('hidapi') on Mac M1 - Solution

### DISCUSSION THREAD

**Discussion by OndrejPistora** [JakubAndrysek/PySpaceMouse#24] [General]
I was getting
_AttributeError: function/symbol 'hid_enumerate' not found in library '<None>': dlsym(RTLD_DEFAULT, hid_enumerate): symbol not found_
on Mac M1
ie easyhid wasnt able to find hidapi library
i installed hidapi using homebrew. it is indeed in _/opt/homebrew/Cellar/hidapi/0.14.0/lib_ but still command
`export DYLD_LIBRARY_PATH=/opt/homebrew/Cellar/hidapi/0.14.0/lib:$DYLD_LIBRARY_PATH`
wasnt helping me fix it. (also as far as I undestood it its temporary until you close terminal so you would need to append it to .zshrc which is not recommended) 
i fixed it by creating symlink
`sudo ln -s /opt/homebrew/Cellar/hidapi/0.14.0/lib/libhidapi.dylib /usr/local/lib/libhidapi.dylib`

**Comment by JakubAndrysek** [comment]
Oh, sorry, I do not have a space mouse near me right now. I will look at it in the future.

**Comment by JakubAndrysek** [comment]
Look at the new [Troubleshooting section](https://spacemouse.kubaandrysek.cz/troubleshooting/#mac-os-m1).

For me, everything works fine.

**Comment by peter-mitrano-ar** [comment]
This topic was address again in #58 , which just merged, but is not yet released. If anyone has questions/concerns, feel free to open a Issue -- I don't normally look at "discussions".
